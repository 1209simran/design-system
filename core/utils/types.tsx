export type BaseProps = {
  className?: string,
  'data-test'?: string,
};

const baseProps = [
  'className',
  'data-test'
];

export const extractBaseProps = (props: Record<string, any>) => {
  const basePropsObj = baseProps.reduce((acc, curr) => {
    return (
      props[curr] ? { ...acc, [curr]: props[curr] } : { ...acc }
    );
  }, {});

  return basePropsObj;
};
