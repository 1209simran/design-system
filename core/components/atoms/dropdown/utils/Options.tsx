export let dropdownOptions: any[] = [];
export const group1: any[] = [];
const group2: any[] = [];

for (let i = 1; i <= 40; i++) {
  group1.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
  });
}

for (let i = 41; i <= 50; i++) {
  group2.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
  });
}

dropdownOptions.push({
  group: true,
  label: 'Group 1',
  items: group1,
});

dropdownOptions.push({
  group: true,
  label: 'Group 2',
  items: group2,
});

export const selectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  selectedStoryOptions.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
    selected: i === 2,
  });
}

export const multiSelectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  multiSelectedStoryOptions.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
    selected: i === 2 || i === 7,
  });
}

export const storyOptions = group1.slice(0, 10);

export const storyWrapOptions = [
  {
    label: 'Design System Dropdown',
    value: 'Design System Dropdown'
  },
  {
    label: 'UI Kit Dropdown',
    value: 'UI Kit Dropdown',
  },
  {
    label: 'Innovaccer Analytics',
    value: 'Innovaccer Analytics'
  }
];

export const subInfoItems = [
  {
    label: 'Option 1',
    value: 'Option 1',
    subInfo: 'subInfo'
  },
  {
    label: 'Option 2',
    value: 'Option 2',
    subInfo: 'subInfo'
  },
  {
    label: 'Option 3',
    value: 'Option 3',
    subInfo: 'subInfo',
  },
];

export const iconItems = [
  {
    label: 'Option 1',
    value: 'Option 1',
    icon: 'events'
  },
  {
    label: 'Option 2',
    value: 'Option 2',
    icon: 'events',
  },
  {
    label: 'Option 3',
    value: 'Option 3',
    icon: 'events',
  },
];
