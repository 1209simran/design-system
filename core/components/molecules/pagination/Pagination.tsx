import * as React from 'react';
import Button from '@/components/atoms/button';
import Text from '@/components/atoms/text';
import Input from '@/components/atoms/input';

import classNames from 'classnames';

export type PaginationType = 'basic' | 'jump';

export interface PaginationProps {
  /**
   * @default "basic"
   */
  type?: PaginationType;
  /**
   * Total number of pages
   */
  totalPages: number;
  /**
   * Current page
   */
  page?: number;
  /**
   *  Callback when page is changed
   */
  onPageChange: (page: number) => void;
}

export const Pagination = (props: PaginationProps) => {
  const {
    type = 'basic',
    totalPages,
    onPageChange
  } = props;

  const [page, setPage] = React.useState<number>(props.page ? props.page : 1);
  const [init, setInit] = React.useState<boolean>(false);

  React.useEffect(() => {
    if (props.page && props.page >= 1 && props.page <= totalPages) setPage(props.page);
  }, [props.page]);

  const wrapperClass = classNames({
    ['Pagination']: true,
    [`Pagination--${type}`]: type,
  });

  const nextButtonWrapperClass = classNames({
    ['Pagination-buttonWrapper']: true,
    ['Pagination-buttonWrapper--next']: true,

  });

  const prevButtonWrapperClass = classNames({
    ['Pagination-buttonWrapper']: true,
    ['Pagination-buttonWrapper--previous']: true,
  });

  React.useEffect(() => {
    if (init && page) onPageChange(page);
  }, [page]);

  const inputChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
    e.preventDefault();

    const val = parseInt(e.target.value.trim(), 10);
    if (!val || (val > 0 && val <= totalPages)) {
      if (!init) setInit(true);
      setPage(val);
    }
  };

  const onClickHandler = (buttonType: 'prev' | 'next' | 'first' | 'last') => {
    setInit(true);
    switch (buttonType) {
      case 'first':
        setPage(1);
        break;
      case 'last':
        setPage(totalPages);
        break;
      case 'prev':
        if (page > 1) setPage(page - 1);
        break;
      case 'next':
        if (page < totalPages) setPage(page + 1);
        break;
    }
  };

  const buttonHelper: string[] = [];
  if (type === 'basic') buttonHelper.push('mx-3');
  else buttonHelper.push('mx-4');

  return (
    <div className={wrapperClass}>
      <div className={prevButtonWrapperClass}>
        <Button
          onClick={() => onClickHandler('first')}
          disabled={page === 1}
          appearance="transparent"
          size="large"
          icon="first_page"
        />
        <div className={['ml-4', ...buttonHelper].join(' ')}>
          <Button
            onClick={() => onClickHandler('prev')}
            disabled={page === 1}
            size="large"
            icon="navigate_before"
          />
        </div>
      </div>
      {type === 'jump' && (
        <div className="Pagination-pageIndex">
          <Input
            name="page"
            type="number"
            size="large"
            onChange={inputChangeHandler}
            value={`${page}`}
          />
          <Text>{` of ${totalPages} pages`}</Text>
        </div>
      )}
      <div className={nextButtonWrapperClass}>
        <div className={['mr-4', ...buttonHelper].join(' ')}>
          <Button
            onClick={() => onClickHandler('next')}
            disabled={page === totalPages}
            size="large"
            icon="navigate_next"
          />
        </div>
        <Button
          onClick={() => onClickHandler('last')}
          disabled={page === totalPages}
          appearance="transparent"
          size="large"
          icon="last_page"
        />
      </div>
    </div >
  );
};

Pagination.displayName = 'Pagination';

export default Pagination;
