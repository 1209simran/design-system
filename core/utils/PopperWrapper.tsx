import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Manager, Reference, Popper } from 'react-popper';

type PositionType =
  | 'auto-start'
  | 'auto'
  | 'auto-end'
  | 'top-start'
  | 'top'
  | 'top-end'
  | 'right-start'
  | 'right'
  | 'right-end'
  | 'bottom-end'
  | 'bottom'
  | 'bottom-start'
  | 'left-end'
  | 'left'
  | 'left-start'
  | undefined;

type actionType = 'click' | 'hover';

enum Offsets {
  Small = '2px',
  Medium = '4px',
  Large = '8px'
}

interface IProps {
  trigger: React.ReactElement<any>;
  placement: PositionType;
  children: React.ReactElement<any>;
  style: React.CSSProperties;
  appendToBody: boolean;
  on?: actionType;
  offset: keyof typeof Offsets;
  closeOnBackdropClick?: boolean;
  hoverable?: boolean;
}

interface IState {
  open: boolean;
  mouseLeaveDelay: number;
  mouseEnterDelay: number;
}

class PopperWrapper extends React.Component<IProps, IState> {
  private triggerRef: React.RefObject<HTMLElement>;
  private popupRef: React.RefObject<HTMLDivElement>;
  private _timer?: number;

  constructor(props: IProps) {
    super(props);

    this.state = {
      open: false,
      mouseLeaveDelay: 50,
      mouseEnterDelay: 0
    };

    this.triggerRef = React.createRef();
    this.popupRef = React.createRef();
  }

  public componentWillUnmount() {
    clearTimeout(this._timer);
    document.removeEventListener('mousedown', this.doesNodeContainClick);
  }

  public handleMouseLeave = (event: React.MouseEvent<HTMLElement>) => {
    const { hoverable = false } = this.props;

    if (hoverable) {
      clearTimeout(this._timer);
      this._timer = window.setTimeout(() => {
        this.setState({ open: false });
        if (this.props.children.props.onMouseLeave) {
          this.props.children.props.onMouseLeave(event);
        }
      }, this.state.mouseLeaveDelay);
    } else {
      this.setState({ open: false });
      if (this.props.children.props.onMouseLeave) {
        this.props.children.props.onMouseLeave(event);
      }
    }
  }

  public handleMouseEnter = (event: React.MouseEvent<HTMLElement>) => {
    const { hoverable = false } = this.props;

    if (hoverable) {
      clearTimeout(this._timer);
      this._timer = window.setTimeout(() => {
        this.setState({ open: true });
        if (this.props.children.props.onMouseEnter) {
          this.props.children.props.onMouseEnter(event);
        }
      }, this.state.mouseEnterDelay);
    } else {
      this.setState({ open: true });
      if (this.props.children.props.onMouseEnter) {
        this.props.children.props.onMouseEnter(event);
      }
    }
  }

  public togglePopper = () => {
    this.setState({ open: !this.state.open });
  }

  public doesNodeContainClick = (event: Event) => {
    if (
      !(
        this.findDOMNode(this.popupRef).contains(event.target as HTMLElement) ||
        this.findDOMNode(this.triggerRef).contains(event.target as HTMLElement)
      )
    ) {
      this.togglePopper();
    }
  }

  public componentDidMount() {
    const { on = 'click', closeOnBackdropClick = true } = this.props;
    const { open } = this.state;

    if (on === 'click' && open && closeOnBackdropClick) {
      document.addEventListener('mousedown', this.doesNodeContainClick);
    }
  }

  public componentDidUpdate() {
    const { on = 'click', closeOnBackdropClick = true } = this.props;
    const { open } = this.state;

    if (on === 'click' && open && closeOnBackdropClick) {
      document.addEventListener('mousedown', this.doesNodeContainClick);
    } else if (on === 'click' && !open && closeOnBackdropClick) {
      document.removeEventListener('mousedown', this.doesNodeContainClick);
    }
  }

  public getTriggerElement(trigger: React.ReactElement<any>, ref: React.Ref<any>, on: actionType) {
    const options =
      on === 'hover'
        ? {
          ref,
          onMouseEnter: this.handleMouseEnter,
          onMouseLeave: this.handleMouseLeave
        }
        : {
          ref,
          onClick: this.togglePopper
        };

    const element = React.cloneElement(
      (
        <span className="d-inline-block">
          {trigger}
        </span>
      ),
      options
      );

    return element;
  }

  public getChildrenElement(
    children: React.ReactElement<any>,
    ref: React.Ref<any>,
    placement: string,
    style: React.CSSProperties
  ) {
    const options = this.props.on === 'hover' ? {
      ref,
      style,
      onMouseEnter: this.handleMouseEnter,
      onMouseLeave: this.handleMouseLeave,
      'data-placement': placement
    } : {
      ref,
      style,
      'data-placement': placement
    };

    const element = React.cloneElement(children, options);
    return element;
  }

  public render() {
    const { trigger, children, placement, appendToBody, on = 'click', offset } = this.props;
    const { open } = this.state;

    return (
      <Manager>
        <Reference innerRef={this.triggerRef}>{({ ref }) => this.getTriggerElement(trigger, ref, on)}</Reference>
        {open &&
          appendToBody &&
          ReactDOM.createPortal(
            (
              /* tslint:disable:no-shadowed-variable */
              <Popper placement={placement} innerRef={this.popupRef}>
                {({ ref, style, placement }) => {
                  const newStyle = offset ? this.getUpdatedStyle(style, placement, offset) : style;
                  return this.getChildrenElement(children, ref, placement, newStyle);
                }}
              </Popper>
            ),
            document.body
          )}
        {open && !appendToBody && (
          <Popper placement={placement} innerRef={this.popupRef}>
            {({ ref, style, placement }) => {
              const newStyle = offset ? this.getUpdatedStyle(style, placement, offset) : style;
              return this.getChildrenElement(children, ref, placement, newStyle);
            }}
          </Popper>
        )}
      </Manager>
    );
  }

  private getUpdatedStyle = (
    oldStyle: React.CSSProperties,
    placement: PositionType,
    offset: keyof typeof Offsets = 'Medium'
  ) => {
    const { style = {} } = this.props;
    const newStyle = { ...style, ...oldStyle };
    const position = placement ? placement.split('-')[0] : placement;
    switch (position) {
      case 'top':
        newStyle.marginBottom = Offsets[offset];
        break;

      case 'bottom':
        newStyle.marginTop = Offsets[offset];
        break;

      case 'left':
        newStyle.marginRight = Offsets[offset];
        break;

      case 'right':
        newStyle.marginLeft = Offsets[offset];
        break;
    }
    return newStyle;
  }

  private findDOMNode = (ref: React.RefObject<HTMLElement>) => {
    return ReactDOM.findDOMNode(ref.current!) as HTMLElement;
  }
}

export default PopperWrapper;
