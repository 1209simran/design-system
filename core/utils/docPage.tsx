// @ts-nocheck

import * as React from 'react';
import { Title, Props, Description, Primary, Story, PropsProps } from '@storybook/addon-docs/blocks';

interface DocPageProps {
  title?: string;
  description?: React.ReactNode;
  props?: PropsProps;
}

export const docPage = (props: DocPageProps) => {
  const {
    title,
    description,
    props,
    stories
  } = props;

  const separatorIndex = title?.indexOf('|');

  return (
    <>
      <Title>{title && separatorIndex !== -1 ? title.slice(separatorIndex + 1) : title}</Title>
      <Description>
        {description}
      </Description>
      {/* <Primary /> */}
      {/* <Story id="switch-state--alert"  /> */}
      <Props {...props} />
    </>
  );
};

export default docPage;
