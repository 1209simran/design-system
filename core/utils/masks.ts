import { Mask } from '@/utils/types';
import { DatePickerProps } from "@/index.type";

type DateFormat = DatePickerProps['inputFormat'];
export const date: Record<DateFormat, Mask> = {
  'dd/mm/yyyy': [/[0123]/, /\d/, '/', /\[01]/, /\d/, '/', /\d/, /\d/, /\d/, /\d/],
  'mm/dd/yyyy': [/[01]/, /\d/, '/', /[0123]/, /\d/, '/', /\d/, /\d/, /\d/, /\d/],
  'yyyy/mm/dd': [/\d/, /\d/, /\d/, /\d/, '/', /[01]/, /\d/, '/', /[0123]/, /\d/],
  'dd-mm-yyyy': [/[0123]/, /\d/, '-', /[01]/, /\d/, '-', /\d/, /\d/, /\d/, /\d/],
  'mm-dd-yyyy': [/[01]/, /\d/, '-', /[0123]/, /\d/, '-', /\d/, /\d/, /\d/, /\d/],
  'yyyy-mm-dd': [/\d/, /\d/, /\d/, /\d/, '-', /[01]/, /\d/, '-', /[0123]/, /\d/],
};
