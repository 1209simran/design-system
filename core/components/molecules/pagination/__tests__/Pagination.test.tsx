import * as React from 'react';
import { shallow } from 'enzyme';
import Pagination, { PaginationProps as Props } from '../Pagination';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';

const pageNumber = [1, 5, 10, 15];
const totalPages = 50;
const paginationType = ['basic', 'jump'];
const FunctionValue = jest.fn();

const mapper: Record<string, any> = {
  type: valueHelper(paginationType, { required: true, iterate: true }),
  page: valueHelper(pageNumber, { required: true, iterate: true }),
  totalPages: valueHelper(totalPages, { required: true }),
  onPageChange: valueHelper(FunctionValue, { required: true }),
};

describe('Pagination component', () => {
  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Pagination
          {...attr}
          totalPages={10}
          onPageChange={jest.fn()}
        />
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});
