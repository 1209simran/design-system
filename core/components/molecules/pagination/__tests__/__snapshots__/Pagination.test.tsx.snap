// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component 
  type: "basic", page: -1, totalPages: 10, onPageChange: "[Function]"
 1`] = `
<div
  className="Pagination Pagination--basic"
>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--previous"
  >
    <Button
      appearance="transparent"
      disabled={true}
      icon="first_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
    <div
      className="ml-4 mx-3"
    >
      <Button
        appearance="basic"
        disabled={true}
        icon="navigate_before"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
  </div>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--next"
  >
    <div
      className="mr-4 mx-3"
    >
      <Button
        appearance="basic"
        disabled={false}
        icon="navigate_next"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
    <Button
      appearance="transparent"
      disabled={false}
      icon="last_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
  </div>
</div>
`;

exports[`Pagination component 
  type: "basic", page: 0, totalPages: 10, onPageChange: "[Function]"
 1`] = `
<div
  className="Pagination Pagination--basic"
>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--previous"
  >
    <Button
      appearance="transparent"
      disabled={true}
      icon="first_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
    <div
      className="ml-4 mx-3"
    >
      <Button
        appearance="basic"
        disabled={true}
        icon="navigate_before"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
  </div>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--next"
  >
    <div
      className="mr-4 mx-3"
    >
      <Button
        appearance="basic"
        disabled={false}
        icon="navigate_next"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
    <Button
      appearance="transparent"
      disabled={false}
      icon="last_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
  </div>
</div>
`;

exports[`Pagination component 
  type: "basic", page: 1, totalPages: 10, onPageChange: "[Function]"
 1`] = `
<div
  className="Pagination Pagination--basic"
>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--previous"
  >
    <Button
      appearance="transparent"
      disabled={true}
      icon="first_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
    <div
      className="ml-4 mx-3"
    >
      <Button
        appearance="basic"
        disabled={true}
        icon="navigate_before"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
  </div>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--next"
  >
    <div
      className="mr-4 mx-3"
    >
      <Button
        appearance="basic"
        disabled={false}
        icon="navigate_next"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
    <Button
      appearance="transparent"
      disabled={false}
      icon="last_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
  </div>
</div>
`;

exports[`Pagination component 
  type: "basic", page: 5, totalPages: 10, onPageChange: "[Function]"
 1`] = `
<div
  className="Pagination Pagination--basic"
>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--previous"
  >
    <Button
      appearance="transparent"
      disabled={false}
      icon="first_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
    <div
      className="ml-4 mx-3"
    >
      <Button
        appearance="basic"
        disabled={false}
        icon="navigate_before"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
  </div>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--next"
  >
    <div
      className="mr-4 mx-3"
    >
      <Button
        appearance="basic"
        disabled={false}
        icon="navigate_next"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
    <Button
      appearance="transparent"
      disabled={false}
      icon="last_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
  </div>
</div>
`;

exports[`Pagination component 
  type: "basic", page: 10, totalPages: 10, onPageChange: "[Function]"
 1`] = `
<div
  className="Pagination Pagination--basic"
>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--previous"
  >
    <Button
      appearance="transparent"
      disabled={false}
      icon="first_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
    <div
      className="ml-4 mx-3"
    >
      <Button
        appearance="basic"
        disabled={false}
        icon="navigate_before"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
  </div>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--next"
  >
    <div
      className="mr-4 mx-3"
    >
      <Button
        appearance="basic"
        disabled={true}
        icon="navigate_next"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
    <Button
      appearance="transparent"
      disabled={true}
      icon="last_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
  </div>
</div>
`;

exports[`Pagination component 
  type: "basic", page: 11, totalPages: 10, onPageChange: "[Function]"
 1`] = `
<div
  className="Pagination Pagination--basic"
>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--previous"
  >
    <Button
      appearance="transparent"
      disabled={false}
      icon="first_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
    <div
      className="ml-4 mx-3"
    >
      <Button
        appearance="basic"
        disabled={false}
        icon="navigate_before"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
  </div>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--next"
  >
    <div
      className="mr-4 mx-3"
    >
      <Button
        appearance="basic"
        disabled={true}
        icon="navigate_next"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
    <Button
      appearance="transparent"
      disabled={true}
      icon="last_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
  </div>
</div>
`;

exports[`Pagination component 
  type: "jump", page: -1, totalPages: 10, onPageChange: "[Function]"
 1`] = `
<div
  className="Pagination Pagination--jump"
>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--previous"
  >
    <Button
      appearance="transparent"
      disabled={true}
      icon="first_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
    <div
      className="ml-4 mx-4"
    >
      <Button
        appearance="basic"
        disabled={true}
        icon="navigate_before"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
  </div>
  <div
    className="Pagination-pageIndex"
  >
    <Input
      name="page"
      onChange={[Function]}
      size="large"
      type="number"
      value="-1"
    />
    <Text
      appearance="default"
      size="regular"
    >
       of 10 pages
    </Text>
  </div>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--next"
  >
    <div
      className="mr-4 mx-4"
    >
      <Button
        appearance="basic"
        disabled={false}
        icon="navigate_next"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
    <Button
      appearance="transparent"
      disabled={false}
      icon="last_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
  </div>
</div>
`;

exports[`Pagination component 
  type: "jump", page: 0, totalPages: 10, onPageChange: "[Function]"
 1`] = `
<div
  className="Pagination Pagination--jump"
>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--previous"
  >
    <Button
      appearance="transparent"
      disabled={true}
      icon="first_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
    <div
      className="ml-4 mx-4"
    >
      <Button
        appearance="basic"
        disabled={true}
        icon="navigate_before"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
  </div>
  <div
    className="Pagination-pageIndex"
  >
    <Input
      name="page"
      onChange={[Function]}
      size="large"
      type="number"
      value=""
    />
    <Text
      appearance="default"
      size="regular"
    >
       of 10 pages
    </Text>
  </div>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--next"
  >
    <div
      className="mr-4 mx-4"
    >
      <Button
        appearance="basic"
        disabled={false}
        icon="navigate_next"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
    <Button
      appearance="transparent"
      disabled={false}
      icon="last_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
  </div>
</div>
`;

exports[`Pagination component 
  type: "jump", page: 1, totalPages: 10, onPageChange: "[Function]"
 1`] = `
<div
  className="Pagination Pagination--jump"
>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--previous"
  >
    <Button
      appearance="transparent"
      disabled={true}
      icon="first_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
    <div
      className="ml-4 mx-4"
    >
      <Button
        appearance="basic"
        disabled={true}
        icon="navigate_before"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
  </div>
  <div
    className="Pagination-pageIndex"
  >
    <Input
      name="page"
      onChange={[Function]}
      size="large"
      type="number"
      value="1"
    />
    <Text
      appearance="default"
      size="regular"
    >
       of 10 pages
    </Text>
  </div>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--next"
  >
    <div
      className="mr-4 mx-4"
    >
      <Button
        appearance="basic"
        disabled={false}
        icon="navigate_next"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
    <Button
      appearance="transparent"
      disabled={false}
      icon="last_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
  </div>
</div>
`;

exports[`Pagination component 
  type: "jump", page: 5, totalPages: 10, onPageChange: "[Function]"
 1`] = `
<div
  className="Pagination Pagination--jump"
>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--previous"
  >
    <Button
      appearance="transparent"
      disabled={false}
      icon="first_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
    <div
      className="ml-4 mx-4"
    >
      <Button
        appearance="basic"
        disabled={false}
        icon="navigate_before"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
  </div>
  <div
    className="Pagination-pageIndex"
  >
    <Input
      name="page"
      onChange={[Function]}
      size="large"
      type="number"
      value="5"
    />
    <Text
      appearance="default"
      size="regular"
    >
       of 10 pages
    </Text>
  </div>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--next"
  >
    <div
      className="mr-4 mx-4"
    >
      <Button
        appearance="basic"
        disabled={false}
        icon="navigate_next"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
    <Button
      appearance="transparent"
      disabled={false}
      icon="last_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
  </div>
</div>
`;

exports[`Pagination component 
  type: "jump", page: 10, totalPages: 10, onPageChange: "[Function]"
 1`] = `
<div
  className="Pagination Pagination--jump"
>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--previous"
  >
    <Button
      appearance="transparent"
      disabled={false}
      icon="first_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
    <div
      className="ml-4 mx-4"
    >
      <Button
        appearance="basic"
        disabled={false}
        icon="navigate_before"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
  </div>
  <div
    className="Pagination-pageIndex"
  >
    <Input
      name="page"
      onChange={[Function]}
      size="large"
      type="number"
      value="10"
    />
    <Text
      appearance="default"
      size="regular"
    >
       of 10 pages
    </Text>
  </div>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--next"
  >
    <div
      className="mr-4 mx-4"
    >
      <Button
        appearance="basic"
        disabled={true}
        icon="navigate_next"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
    <Button
      appearance="transparent"
      disabled={true}
      icon="last_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
  </div>
</div>
`;

exports[`Pagination component 
  type: "jump", page: 11, totalPages: 10, onPageChange: "[Function]"
 1`] = `
<div
  className="Pagination Pagination--jump"
>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--previous"
  >
    <Button
      appearance="transparent"
      disabled={false}
      icon="first_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
    <div
      className="ml-4 mx-4"
    >
      <Button
        appearance="basic"
        disabled={false}
        icon="navigate_before"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
  </div>
  <div
    className="Pagination-pageIndex"
  >
    <Input
      name="page"
      onChange={[Function]}
      size="large"
      type="number"
      value="11"
    />
    <Text
      appearance="default"
      size="regular"
    >
       of 10 pages
    </Text>
  </div>
  <div
    className="Pagination-buttonWrapper Pagination-buttonWrapper--next"
  >
    <div
      className="mr-4 mx-4"
    >
      <Button
        appearance="basic"
        disabled={true}
        icon="navigate_next"
        iconAlign="left"
        onClick={[Function]}
        size="large"
        tabIndex={0}
      />
    </div>
    <Button
      appearance="transparent"
      disabled={true}
      icon="last_page"
      iconAlign="left"
      onClick={[Function]}
      size="large"
      tabIndex={0}
    />
  </div>
</div>
`;
